apps/judicial-system/backend/src/app/modules/aws-s3/awsS3.service.ts

```diff
@@ -37,72 +45,66 @@ const formatS3Key = (caseType: CaseType, key: string) => {
 
 @Injectable()
 export class AwsS3Service {
-  private readonly s3: S3
+  private readonly s3Client: S3Client
 
   constructor(
     @Inject(awsS3ModuleConfig.KEY)
     private readonly config: ConfigType<typeof awsS3ModuleConfig>,
     @Inject(LOGGER_PROVIDER) private readonly logger: Logger,
   ) {
-    this.s3 = new S3({ region: this.config.region })
+    this.s3Client = new S3Client({ region: this.config.region })
   }
 
-  createPresignedPost(
+  async createPresignedPost(
     caseType: CaseType,
     key: string,
     type: string,
-  ): Promise<S3.PresignedPost> {
-    return new Promise((resolve, reject) => {
-      this.s3.createPresignedPost(
-        {
-          Bucket: this.config.bucket,
-          Expires: this.config.timeToLivePost,
-          Fields: {
-            key: formatS3Key(caseType, key),
-            'content-type': type,
-            'Content-Disposition': 'inline',
-          },
-        },
-        (err, data) => {
-          if (err) {
-            reject(err)
-          } else {
-            resolve(data)
-          }
-        },
-      )
+  ): Promise<{ url: string; fields: Record<string, string> }> {
+    const { url, fields } = await createPresignedPost(this.s3Client, {
+      Bucket: this.config.bucket,
+      Key: formatS3Key(caseType, key),
+      Conditions: [
+        ['content-length-range', 0, 10485760], // 10 MB
```

Comment:
Is there a hard limit on the file size? We are dealing with court documents and they may get bigger than this.

Reply:
Yes, this is a hard limit. What size do you suggest?

Reply:
The files we upload to S3 are later uploaded to the district courts. The API we are working against there currently has a limit of 100 MB. Our limit needs to reflect that. Note that the bucket we are writing to in S3 is not the same as any buckets used by island.is. Until now, we have not been restricted to any limit as far as I know.

AWS S3 itself has a maximum object size limit:
	•	Single PUT operation: The maximum size for a single PUT operation (which includes uploads via pre-signed URLs) is 5 GB.
	•	Multipart upload: For larger files, you can use the multipart upload API to upload objects up to 5 TB in size.

Reply:
Thank you.